### ISSUE TITLE
Jena version mismatch

### ISSUE THREAD

**Issue by ajs6f** [trellis-ldp/trellis#342]
We're using Jena 3.9.0 to support `trellis-rdfa` but Commons RDF is still stuck on 3.5.0. I've created [a Commons ticket](https://issues.apache.org/jira/browse/COMMONSRDF-80) for the long-term fix, but for now, trying to boot Trellis is giving me:
```
2018-12-29 11:41:29,742 ERROR [stderr] (main)     Caused by: java.util.ServiceConfigurationError: org.apache.commons.rdf.api.RDF: Provider org.apache.commons.rdf.jena.JenaRDF could not be instantiated
2018-12-29 11:41:29,742 ERROR [stderr] (main)     Caused by: java.util.ServiceConfigurationError: org.apache.jena.sys.JenaSubsystemLifecycle: Provider org.apache.jena.riot.system.InitRIOT not a subtype"}}
```
which points to the version mismatch.

**Comment by ajs6f** [comment]
PR in to Commons, but who knows how long it will take to cut a release?

**Comment by acoburn** [comment]
I think the issue is that I've changed the jena dependency from `jena-osgi` to `jena-arq`, etc. Really, the -osgi artifact is great for OSGi environments, but there's no need for it outside of OSGi. So, the build code currently swaps out the jena-osgi dependency for jena-arq -- I'd suggest you try doing the same in the T-C* code.

Ideally, the Commons code would start using the jena-arq dependency instead. Anyone who wants to use this in an OSGi context would be using the osgi dependency anyway.

**Comment by ajs6f** [comment]
I have tried that, and it doesn't work. Why? Good question. I'm trying to find out.

**Comment by ajs6f** [comment]
Hang on-- did you exclude it too? So you aren't pulling  in`-osgi` transitively?

**Comment by acoburn** [comment]
I completely excluded it, directly and transitively.

**Comment by ajs6f** [comment]
Hm, so did I, so that isn't it.

**Comment by acoburn** [comment]
Even better would be for me to update the generated POM files so that the dependencies are excluded there.

**Comment by ajs6f** [comment]
Already excluded-- that isn't the problem (whatever the problem is!).

**Comment by ajs6f** [comment]
Yes, better to go as upstream as possible. Until COMMONS-80 get resolved, we should do this (`-osgi` => `-arq`) for ourselves.

**Comment by ajs6f** [comment]
Incidentally, I'm failing the integration tests with an interesting message:
```
java.lang.NoClassDefFoundError: Could not initialize class org.trellisldp.vocabulary.LDP
```
Why that class of all classes? Investigating...

**Comment by acoburn** [comment]
Yes, I saw that, too. That's failing b/c the service loader can't find an RDF implementation. I'm working on a fix, which should be ready shortly

**Comment by ajs6f** [comment]
Ah, that makes sense. Do I need to manually include a file in `META-INF/services`?

**Comment by acoburn** [comment]
You shouldn't need to. The issue (I think) is that when `org.apache.commons.rdf.jena.JenaRDF` tries to initialize (via the ServiceLoader), it fails b/c of some class-not-found errors (due to excluding `jena-osgi`). In principle, if you also _include_ `jena-arq`, it should work.

But I have a fix for this at the Trellis layer (building and testing now)

**Comment by acoburn** [comment]
I'll reopen this until you can verify that this achieves the desired result. (I've confirmed this in a local build of T-C*)
